### PR TITLE
Straighten out localization in the scraper.

### PR DESCRIPTION
**Pull request by jamescowens**
Addresses issue #1470.

I also added code to turn off the NN housekeeping thread completely if the -disablenn argument is passed. This might be useful for exchanges that have no use running any portion of the scraper/NN code at all.

### CODE REVIEW

File: src/scraper/scraper.cpp [hunk 1]
```diff
@@ -708,7 +708,7 @@ void Scraper(bool bSingleShot)
     {
         pathDataDir = GetDataDir();
         // This is necessary to maintain compatibility with Windows.
-        pathDataDir.imbue(std::locale(std::locale(), new std::codecvt_utf8_utf16<wchar_t>()));
+        // pathDataDir.imbue(std::locale(std::locale(), new std::codecvt<wchar_t, char, std::mbstate_t>));
```
Comment (denravonska) [inline src/scraper/scraper.cpp]: Consider removing these and their comments if we don't need them anymore.

**Review by denravonska — APPROVED** [review]
utACK.

File: src/scraper/scraper.cpp (hunk 1 above)
Comment (jamescowens) [reply src/scraper/scraper.cpp]: Yeah I forgot! :)

Comment (jamescowens) [reply src/scraper/scraper.cpp]: Done.

**Review by cyrossignol — APPROVED** [review]
Looks like this solves the issue for the data directory with an `ó` character in the path as reported. Inspired by @barton2526, I tried `ḨE_H͠UNGER̢S̴ ` and `世`--these didn't work, so I guess we don't have full Unicode path support yet.

The `-disablenn` flag works as expected--no NeuralNetwork thread started.

**Comment by jamescowens** [comment]
Full Unicode support is going to require more extensive changes in the rest of the app, because we are using the default locale everywhere, or in some instances classic, and not all operating systems actually have UTF32 support for filesystem paths anyway... we are probably in a good enough place with this now.

**Comment by jamescowens** [comment]
Merging.

**Pull request merged**